### PR TITLE
Extend search

### PR DESCRIPTION
**Pull request by jernejpregelj**
Extending search with filters.

Depends on https://github.com/oceanprotocol/aquarius/issues/223

Current state:

Screen Shot 2019-09-13 at 13 21 32

### CODE REVIEW

**Comment by kremalicious** [comment]
quick prototype in https://github.com/oceanprotocol/commons/pull/183/commits/f2024384f07f6d388d3ab63bc9d44bb56c1ee598 with a filter sidebar pattern, meaning the filters modify the primary text search:

Screen Shot 2019-08-01 at 15 59 29

How that filter sidebar should work:

1. only show entries which are found in the search results, possibly with number of results shown after each item
2. clicking on one item fires new search with that item
3. multiple items can be selected and initiate a new search

Would be awesome to get this filter bar quicker by not firing new search and filtering on client side through existing results. But guess that's not so feasible because of pagination and all.

File: client/src/routes/Search/Filters.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    steps[2].fields &&
+    steps[2].fields.license &&
+    steps[2].fields.license.options
+
+const filters = [
+    { label: labelCategories, items: optionsCategories },
+    { label: labelLicense, items: optionsLicense }
+]
+
+export default class Sidebar extends PureComponent<{
```
Comment (kremalicious) [inline client/src/routes/Search/Filters.tsx]:
```suggestion
export default class Filters extends PureComponent<{
```

whoops, should be `Filters`

**Comment by jernejpregelj** [comment]
There are problems with search combinations described in https://github.com/oceanprotocol/aquarius/issues/211

**Comment by kremalicious** [comment]
In principle this advanced search works but it behaves inconsistent once pagination is involved. To solve this we require consolidated metadata to be part of Aquarius search response. 

As a start, we need a consolidated array of categories & licenses all returned search results are in, across pagination.

```
{
    "results": [],
    "metadata": {
        "categories": [
            {
                "name": "Category",
                "count": 3
            }
        ],
        "license": [
            {
                "name": "Public Domain",
                "count": 3
            }
        ]
    },
    "page": 1,
    "total_pages": 1520,
    "total_results": 1520
}
```

**Pull request closed without merging**